src/modules/RA/Viewpoint/views/List.js

```diff
@@ -1,13 +1,77 @@
 import React from 'react';
-import { List } from 'react-admin';
+import {
+  Filter,
+  List,
+  SelectArrayInput,
+  SelectInput,
+  TextInput,
+} from 'react-admin';
 
 import CommonBulkActionButtons from '../../../../components/react-admin/CommonBulkActionButtons';
 import GridList from '../../../../components/react-admin/GridList';
+import { fetchFilterOptions } from '../../ra-modules';
+import { connectAppProvider } from '../../../../components/AppProvider';
 
+
+const ListFilters = props => {
+  const [selectInputs, setSelectInputs] = React.useState([]);
+
+  const setFilters = async () => {
+    const filterOptions = await fetchFilterOptions();
+    const terraOppSearchableProperties = props.properties;
+
+    const selectInputList = Object.keys(terraOppSearchableProperties).reduce(
+      (filters, filterName) => {
+        if (filterName in filterOptions) {
+          const choices = filterOptions[filterName].map(value => ({ name: value, id: value }));
+          if (terraOppSearchableProperties[filterName].type === 'many') {
+            filters.push(
```

Comment:
We should avoid to mix business logic and JSX. Data processing should be separated from rendering concerns.

First: Creating a variable with all requirement for rendering
Then: Render UI elements from previously created variable.

Also, rendering may be managed by a proper dedicaed component.